tools/twix/src/repository_parameters.rs

```diff
@@ -32,46 +32,61 @@ impl RepositoryParameters {
 
     pub fn write(&self, address: &str, path: String, value: Value) {
         let repository = self.repository.clone();
-        let Ok(head_id) = self
-            .head_id_from_address(address)
+        let Ok(hardware_ids) = self
+            .hardware_ids_from_address(address)
         else {
             error!("failed to get head ID from address {address}");
             return
         };
         self.runtime.spawn(async move {
-            let mut stored_value = repository
-                .read_configuration(&head_id)
+            let stored_complete_parameter_tree: Value = parameters::directory::deserialize(
+                repository.root_directory(),
+                &hardware_ids.body_id,
+                &hardware_ids.head_id,
+            )
+            .await
+            .unwrap_or_default();
+
+            // value is just the "leaf" of the "path", make the tree structure to diff/ merge against complete parameter tree.
```

Comment:
This comment explains ***what*** happens, which the code already expresses. I propose to either remove the comment to not violate DRY or explain the ***why***.

Reply:
Makes sense, the variable naming could be clearer if needed instead of that comment.